tket/src/Circuit/basic_circ_manip.cpp

```diff
@@ -384,17 +414,14 @@ void Circuit::add_qubit(const Qubit& id, bool reject_dups) {
           "A unit with ID \"" + id.repr() + "\" already exists");
     } else if (found->type() == UnitType::Qubit) {
       return;
-    } else {
-      throw CircuitInvalidity(
-          "A bit with ID \"" + id.repr() + "\" already exists");
     }
   }
   opt_reg_info_t reg_info = get_reg_info(id.reg_name());
   register_info_t correct_info = {UnitType::Qubit, id.reg_dim()};
-  if (reg_info && !(reg_info.value() == correct_info))
-    throw CircuitInvalidity(
-        "Cannot add qubit with ID \"" + id.repr() +
-        "\" as register is not compatible");
+
+  // Cannot add qubit with ID id.repr() ss register is not compatible
```

Comment:
```suggestion
  // Cannot add qubit with this ID as register is not compatible
```

Reply:
done in [suggested changes I](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)  [e06543a](https://github.com/CQCL/tket/pull/743/commits/e06543a3343027c8ff962d77fda395d3b22d9f96)